apis/exoscale/v1/iam_types.go

```diff
@@ -67,7 +68,7 @@ type IAMKeyObservation struct {
 	KeyID string `json:"keyID,omitempty"`
 
 	// RoleID is the observed unique ID as generated by exoscale.com.
-	RoleID string `json:"roleID,omitempty"`
+	RoleID exoscalesdk.UUID `json:"roleID,omitempty"`
```

Comment:
This might not work with current versions, but hopefully I am wrong

Reply:
please @TheBigLee add tests for this case

Reply:
I've added a test for this